frotz.py

```diff
@@ -143,19 +167,19 @@ def read_file(self, file_path):
 
     def open_url(self, url):
         """Ask the journal to open an URL for us."""
-        from sugar import profile
+        from sugar3 import profile
         from shutil import rmtree
-        from sugar.datastore import datastore
-        from sugar.activity.activity import show_object_in_journal
+        from sugar3.datastore import datastore
+        from sugar3.activity.activity import show_object_in_journal
         from tempfile import mkdtemp
         tmpfolder = mkdtemp('.tmp', 'url', os.path.join(self.get_activity_root(), 'instance'))
         tmpfilepath = os.path.join(tmpfolder, 'url')
         try:
             tmpfile = open(tmpfilepath, 'w')
             tmpfile.write(url)
             tmpfile.close()
-            os.chmod(tmpfolder, 0755)
-            os.chmod(tmpfilepath, 0755)
+            os.chmod(tmpfolder, 0o755)
+            os.chmod(tmpfilepath, 0o755)
```

Comment:
out of curiosity and due to security reasons, maybe we should not make a temp file executable for the world?

Reply:
Yeah I was also curious to know why they are doing this. Isn't there any other way? :thinking:

Reply:
Change it to 644 perhaps

Reply:
Great , done in 607dd6f